test/unit/nic_test.rb

```diff
@@ -401,6 +401,25 @@ class DisallowedTestNic < Nic::Base
     assert_equal(interface.domain, domain)
   end
 
+  test 'new nic name containing existing subdomain should set nic domain correctly' do
```

Comment:
I think you can do without db calls for most of these tests - creating a host is expensive.

```ruby

    test 'new nic name containing existing subdomain should set nic domain correctly' do
      nic_name = 'hostname.sub.bigdomain'
      interface = FactoryGirl.build_stubbed(:nic_managed, :name => nic_name)
      subdomain = FactoryGirl.create(:domain, :name => 'sub.bigdomain')
      Domain.expects(:find_by_name).with(subdomain.name).returns(subdomain)
      interface.send(:normalize_name)
      assert_equal subdomain, interface.domain
    end

    test 'new nic name containing non-existing subdomain should not set nic domain' do
      nic_name = 'hostname.undefined-subdomain.bigdomain'
      interface = FactoryGirl.build_stubbed(:nic_managed, :name => nic_name)
      Domain.expects(:find_by_name).
        with('undefined-subdomain.bigdomain').
        returns(nil)
      interface.send(:normalize_name)
      refute interface.domain
    end
```

I realize `.send` is a smell on unit tests. It's there because `normalize_name` shouldn't be part of Nic::Interface, but that's another story and I don't think it should be refactored to fix this ticket.